File: frontend/src/types/preferences.ts
export interface PreferenceSnapshotRequest {
  user_id: string
  liked_poi_ids: string[]
  disliked_poi_ids: string[]
  city: string
}

export interface PreferenceSnapshot {
  user_id: string
  liked_poi_ids: string[]
  disliked_poi_ids: string[]
  tag_weights: Record<string, number>
  category_weights: Record<string, number>
  keyword_weights: Record<string, number>
  source: "ugc_feed_mock"
}

File: backend/app/schemas/preferences.py
from pydantic import BaseModel, Field


class PreferenceSnapshotRequest(BaseModel):
    user_id: str
    liked_poi_ids: list[str] = Field(default_factory=list)
    disliked_poi_ids: list[str] = Field(default_factory=list)
    city: str = "shanghai"


class PreferenceSnapshot(BaseModel):
    user_id: str
    liked_poi_ids: list[str] = Field(default_factory=list)
    disliked_poi_ids: list[str] = Field(default_factory=list)
    tag_weights: dict[str, float] = Field(default_factory=dict)
    category_weights: dict[str, float] = Field(default_factory=dict)
    keyword_weights: dict[str, float] = Field(default_factory=dict)
    source: str = "ugc_feed_mock"
